pkgs/applications/misc/double-commander/default.nix

```diff
@@ -0,0 +1,95 @@
+{
+  stdenv, lib
+  , fetchFromGitHub
+  , fpc, lazarus
+  , which, autoPatchelfHook
+  , gtk2, dbus
+}:
+
+stdenv.mkDerivation rec {
+  pname = "doublecmd";
+  version = "1.0.2";
+
+  src = fetchFromGitHub {
+    owner = "doublecmd";
+    repo = "doublecmd";
+    rev = "v${version}";
+    sha256 = "sha256-nX5EZekYF9saM0QaN9caCnsRDIqPYNX/BNu6Bb5lDYQ=";
+  };
+
+  nativeBuildInputs = [
+    fpc
+    lazarus
+    which
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    gtk2
```

Comment:
I don't think we should merge this PR as it currently stands.

Reply:
@aanderse Can you please give a reason.

Reply:
Sure! I apologize because I thought it more widely understood that `gtk2` is unmaintained software. We shouldn't be adding new software which depends in it. Please build with `gtk3` instead.

Reply:
doublecmd doesn't support gtk3 : https://github.com/doublecmd/doublecmd/blob/1c4d8d0b2014c3f9a339dc134eda8f556f9182de/build.sh#L15

And Lazarus by default targets gtk2 : https://github.com/NixOS/nixpkgs/blob/a0dbe47318bbab7559ffbfa7c4872a517833409f/pkgs/development/compilers/fpc/lazarus.nix#L41

Doublecmd does apparently build on qt and it seems that the Lazarus package supports it with an optional flag so maybe worth a try.

I know gtk2 is "old" and all but the thing is lots of applications still depend on it, notably GIMP...
IMHO this kind of issues should be taken upstream, there's not much to do about it in nixpkgs...

Reply:
Yes, but there is a difference between adding new software and maintaining software we already have packaged. If upstream offers no alternative to `gtk2` it implies the software isn't maintained... but as you have said upstream **does** offer alternative builds ... so I guess this is not as bad as I assumed.

One correction though... `gtk2` isn't just old, it is unmaintained. So can you please try to build with `qt` so we don't add more **direct** dependencies on `gtk2`?

Reply:
Built with qt in 2c7c7e2100a17d55cdcba96772382d1db062cd4b .
However it *still* requires gtk2, otherwise it fails to link, presumably due to limitations in Lazarus? (I haven't used anything related to Pascal in 20 years)
Also it takes **one minute** to start (no CPU usage so who knows what it's doing), making it unusable.
So nothing was gained...

Reply:
Well... you tried at least :man_shrugging: I'm sorry for the hassle :disappointed:

Reply:
Just checked Debian and the qt package doesn't require gtk2 https://packages.debian.org/sid/doublecmd-qt
so maybe there's some way, but I sure don't know what it is :man_shrugging: